src/main/java/org/humancellatlas/ingest/schemas/schemascraper/impl/SchemaScraperImpl.java

```diff
@@ -0,0 +1,53 @@
+package org.humancellatlas.ingest.schemas.schemascraper.impl;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.dataformat.xml.JacksonXmlModule;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import lombok.NonNull;
+import org.humancellatlas.ingest.schemas.Schema;
+import org.humancellatlas.ingest.schemas.schemascraper.SchemaScraper;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Created by rolando on 19/04/2018.
+ *
+ * Scrapes schemas from an s3 bucket's default XML file-list page
+ *
+ */
+@Service
+public class SchemaScraperImpl implements SchemaScraper {
+    private final @NonNull RestTemplate restTemplate;
+    private final @NonNull XmlMapper xmlMapper;
+
+    public SchemaScraperImpl(){
+        this.restTemplate = new RestTemplate(new HttpComponentsClientHttpRequestFactory());
```

Comment:
We can probably externalize this into a config file so that this class only refers to it through `@Autowired`. Better yet, inject `RestTemplateBuilder` here so that we can use `@RestClientTest` with `MockRestServiceServer` in our tests.